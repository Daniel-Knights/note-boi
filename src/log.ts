import { debug, error, info, trace, warn } from '@tauri-apps/plugin-log';

/** Redirects all logs to the log file */
export function initLogger() {
  forwardConsole('log', trace);
  forwardConsole('debug', debug);
  forwardConsole('info', info);
  forwardConsole('warn', warn);
  forwardConsole('error', error);
}

export function forwardConsole(
  fnName: 'log' | 'debug' | 'info' | 'warn' | 'error',
  logger: (message: string) => Promise<void>
) {
  const original = console[fnName];

  console[fnName] = (message: unknown) => {
    original(message);

    if (message instanceof Error) {
      logger(message.name);
      logger(message.message);
      logger(message.stack?.replaceAll(/\n/g, '\n    ') || '');

      return;
    }

    if (typeof message === 'object') {
      logger(JSON.stringify(message));
    }

    logger(String(message));
  };
}
