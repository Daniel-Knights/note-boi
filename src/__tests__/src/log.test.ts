import { forwardConsole, initLogger } from '../../log';

const originalConsole = { ...console };

afterEach(() => {
  console.log = originalConsole.log;
  console.debug = originalConsole.debug;
  console.info = originalConsole.info;
  console.warn = originalConsole.warn;
  console.error = originalConsole.error;
});

describe('Log', () => {
  const mockLogger = vi.fn();

  it('initLogger', () => {
    console.log = mockLogger;
    console.debug = mockLogger;
    console.info = mockLogger;
    console.warn = mockLogger;
    console.error = mockLogger;

    assert.strictEqual(console.log, mockLogger);
    assert.strictEqual(console.debug, mockLogger);
    assert.strictEqual(console.info, mockLogger);
    assert.strictEqual(console.warn, mockLogger);
    assert.strictEqual(console.error, mockLogger);

    initLogger();

    assert.notStrictEqual(console.log, mockLogger);
    assert.notStrictEqual(console.debug, mockLogger);
    assert.notStrictEqual(console.info, mockLogger);
    assert.notStrictEqual(console.warn, mockLogger);
    assert.notStrictEqual(console.error, mockLogger);
  });

  describe('forwardConsole', () => {
    it('console.log', () => {
      forwardConsole('log', mockLogger);

      console.log('test message');

      expect(mockLogger).toHaveBeenCalledOnce();
      expect(mockLogger).toHaveBeenCalledWith('test message');
    });

    it('console.error', () => {
      const error = new Error('test error');

      forwardConsole('error', mockLogger);

      console.error(error);

      expect(mockLogger).toHaveBeenCalledTimes(3);
      expect(mockLogger).toHaveBeenCalledWith(error.name);
      expect(mockLogger).toHaveBeenCalledWith(error.message);
      expect(mockLogger).toHaveBeenCalledWith(
        error.stack?.replaceAll(/\n/g, '\n    ') || ''
      );
    });

    it('console.info', () => {
      const obj = { key: 'value' };

      forwardConsole('info', mockLogger);

      console.info(obj);

      expect(mockLogger).toHaveBeenCalledTimes(2);
      expect(mockLogger).toHaveBeenCalledWith(JSON.stringify(obj));
      expect(mockLogger).toHaveBeenCalledWith(String(obj));
    });

    it('console.warn', () => {
      forwardConsole('warn', mockLogger);

      console.warn(123);

      expect(mockLogger).toHaveBeenCalledOnce();
      expect(mockLogger).toHaveBeenCalledWith('123');
    });

    it('console.debug', () => {
      const message = 'test message';

      forwardConsole('debug', mockLogger);

      console.debug(message);

      expect(mockLogger).toHaveBeenCalledOnce();
      expect(mockLogger).toHaveBeenCalledWith(message);
    });
  });
});
