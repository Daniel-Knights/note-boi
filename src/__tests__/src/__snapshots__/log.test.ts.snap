// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Log > forwardConsole > console.debug > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > forwardConsole > console.debug > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > forwardConsole > console.debug > Storage.get('THEME') 1`] = `null`;

exports[`Log > forwardConsole > console.debug > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > forwardConsole > console.debug > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > forwardConsole > console.debug > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > forwardConsole > console.debug > allCalls 1`] = `[]`;

exports[`Log > forwardConsole > console.debug > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > forwardConsole > console.debug > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > forwardConsole > console.debug > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > forwardConsole > console.debug > openedPopup 1`] = `undefined`;

exports[`Log > forwardConsole > console.debug > selectedTheme 1`] = `"System"`;

exports[`Log > forwardConsole > console.debug > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > forwardConsole > console.debug > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`Log > forwardConsole > console.error > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > forwardConsole > console.error > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > forwardConsole > console.error > Storage.get('THEME') 1`] = `null`;

exports[`Log > forwardConsole > console.error > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > forwardConsole > console.error > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > forwardConsole > console.error > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > forwardConsole > console.error > allCalls 1`] = `[]`;

exports[`Log > forwardConsole > console.error > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > forwardConsole > console.error > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > forwardConsole > console.error > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > forwardConsole > console.error > openedPopup 1`] = `undefined`;

exports[`Log > forwardConsole > console.error > selectedTheme 1`] = `"System"`;

exports[`Log > forwardConsole > console.error > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > forwardConsole > console.error > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`Log > forwardConsole > console.info > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > forwardConsole > console.info > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > forwardConsole > console.info > Storage.get('THEME') 1`] = `null`;

exports[`Log > forwardConsole > console.info > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > forwardConsole > console.info > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > forwardConsole > console.info > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > forwardConsole > console.info > allCalls 1`] = `[]`;

exports[`Log > forwardConsole > console.info > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > forwardConsole > console.info > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > forwardConsole > console.info > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > forwardConsole > console.info > openedPopup 1`] = `undefined`;

exports[`Log > forwardConsole > console.info > selectedTheme 1`] = `"System"`;

exports[`Log > forwardConsole > console.info > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > forwardConsole > console.info > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`Log > forwardConsole > console.log > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > forwardConsole > console.log > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > forwardConsole > console.log > Storage.get('THEME') 1`] = `null`;

exports[`Log > forwardConsole > console.log > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > forwardConsole > console.log > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > forwardConsole > console.log > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > forwardConsole > console.log > allCalls 1`] = `[]`;

exports[`Log > forwardConsole > console.log > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > forwardConsole > console.log > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > forwardConsole > console.log > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > forwardConsole > console.log > openedPopup 1`] = `undefined`;

exports[`Log > forwardConsole > console.log > selectedTheme 1`] = `"System"`;

exports[`Log > forwardConsole > console.log > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > forwardConsole > console.log > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`Log > forwardConsole > console.warn > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > forwardConsole > console.warn > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > forwardConsole > console.warn > Storage.get('THEME') 1`] = `null`;

exports[`Log > forwardConsole > console.warn > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > forwardConsole > console.warn > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > forwardConsole > console.warn > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > forwardConsole > console.warn > allCalls 1`] = `[]`;

exports[`Log > forwardConsole > console.warn > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > forwardConsole > console.warn > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > forwardConsole > console.warn > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > forwardConsole > console.warn > openedPopup 1`] = `undefined`;

exports[`Log > forwardConsole > console.warn > selectedTheme 1`] = `"System"`;

exports[`Log > forwardConsole > console.warn > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > forwardConsole > console.warn > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`Log > initLogger > KeyStore.getKey() 1`] = `undefined`;

exports[`Log > initLogger > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Log > initLogger > Storage.get('THEME') 1`] = `null`;

exports[`Log > initLogger > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Log > initLogger > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Log > initLogger > Storage.get('USERNAME') 1`] = `null`;

exports[`Log > initLogger > allCalls 1`] = `[]`;

exports[`Log > initLogger > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Log > initLogger > mockDb 1`] = `
{
  "deletedNoteIds": Set {},
  "encryptedNotes": [
    {
      "content": "content",
      "id": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
      "timestamp": 958201892455,
    },
    {
      "content": "content",
      "id": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
      "timestamp": 811831863971,
    },
    {
      "content": "content",
      "id": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
      "timestamp": 739056902017,
    },
    {
      "content": "content",
      "id": "6b24cde1-6118-4277-9791-8eab3c71cabb",
      "timestamp": 157836992432,
    },
    {
      "content": "content",
      "id": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
      "timestamp": 1066734006578,
    },
    {
      "content": "content",
      "id": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
      "timestamp": 1642640862249,
    },
    {
      "content": "content",
      "id": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
      "timestamp": 242282290063,
    },
    {
      "content": "content",
      "id": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
      "timestamp": 550235579146,
    },
    {
      "content": "content",
      "id": "b9754a63-843c-4f59-9aea-9a374aa1603d",
      "timestamp": 1533441892487,
    },
    {
      "content": "content",
      "id": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
      "timestamp": 1650008403223,
    },
  ],
  "users": {
    "d": "1",
  },
}
`;

exports[`Log > initLogger > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Log > initLogger > openedPopup 1`] = `undefined`;

exports[`Log > initLogger > selectedTheme 1`] = `"System"`;

exports[`Log > initLogger > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNoteIds": {
    "clear": [Function],
    "deleted": Set {},
    "edited": Set {},
    "new": "",
    "set": [Function],
    "size": 0,
    "store": [Function],
  },
  "username": "",
}
`;

exports[`Log > initLogger > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;
